typescript/cli/src/config/ism.ts

```diff
@@ -87,116 +88,206 @@ export async function createIsmConfig(
+      return createTrustedRelayerConfig(context);
+    default:
+      throw new Error(`Invalid ISM type: ${moduleType}}`);
   }
-
-  throw new Error(`Invalid ISM type: ${moduleType}`);
 }
 
-export async function createMultisigConfig(
-  type: IsmType.MERKLE_ROOT_MULTISIG | IsmType.MESSAGE_ID_MULTISIG,
-): Promise<MultisigIsmConfig> {
-  const thresholdInput = await input({
-    message: 'Enter threshold of validators (number) for multisig ISM',
-  });
-  const threshold = parseInt(thresholdInput, 10);
+export const createMerkleRootMultisigConfig = withOutputContext(
+  async (): Promise<MultisigIsmConfig> => {
+    const thresholdInput = await input({
+      message:
+        'Enter threshold of validators (number) for merkle root multisig ISM',
+    });
+    const threshold = parseInt(thresholdInput, 10);
 
-  const validatorsInput = await input({
-    message:
-      'Enter validator addresses (comma separated list) for multisig ISM',
-  });
-  const validators = validatorsInput.split(',').map((v) => v.trim());
-  return {
-    type,
-    threshold,
-    validators,
-  };
-}
+    const validatorsInput = await input({
+      message:
+        'Enter validator addresses (comma separated list) for merkle root multisig ISM',
+    });
+    const validators = validatorsInput.split(',').map((v) => v.trim());
+    return {
+      type: IsmType.MERKLE_ROOT_MULTISIG,
+      threshold,
+      validators,
+    };
+  },
+  IsmType.MERKLE_ROOT_MULTISIG,
+);
 
-export async function createTrustedRelayerConfig(
-  context: CommandContext,
-): Promise<TrustedRelayerIsmConfig> {
-  const relayer = await detectAndConfirmOrPrompt(
-    async () => context.signer?.getAddress(),
-    'For trusted relayer ISM, enter',
-    'relayer address',
-    'signer',
-  );
-  return {
-    type: IsmType.TRUSTED_RELAYER,
-    relayer,
-  };
-}
+export const createMessageIdMultisigConfig = withOutputContext(
+  async (): Promise<MultisigIsmConfig> => {
+    const thresholdInput = await input({
+      message:
+        'Enter threshold of validators (number) for message ID multisig ISM',
+    });
+    const threshold = parseInt(thresholdInput, 10);
 
-export async function createAggregationConfig(
-  context: CommandContext,
-): Promise<AggregationIsmConfig> {
-  const isms = parseInt(
-    await input({
-      message: 'Enter the number of ISMs to aggregate (number)',
-    }),
-    10,
-  );
-
-  const threshold = parseInt(
-    await input({
-      message: 'Enter the threshold of ISMs for verification (number)',
-    }),
-    10,
-  );
-
-  const modules: Array<IsmConfig> = [];
-  for (let i = 0; i < isms; i++) {
-    modules.push(await createIsmConfig(context));
-  }
-  return {
-    type: IsmType.AGGREGATION,
-    modules,
-    threshold,
-  };
-}
+    const validatorsInput = await input({
+      message:
+        'Enter validator addresses (comma separated list) for message ID multisig ISM',
+    });
+    const validators = validatorsInput.split(',').map((v) => v.trim());
```

Comment:
should we also validate the addresses here?

Reply:
agreed– added a check here to do so